lisp/builtins.go

```diff
@@ -1489,12 +1489,63 @@ func builtinSlice(env *LEnv, args *LVal) *LVal {
 	if i > j {
 		return env.Errorf("end before start")
 	}
+
+	// Create an intermediate sliced list with a similar type
```

Comment:
is this the source of the bulk of the performance gains?

Reply:
Ah. No. I don't have any benchmarks for the slice function in particular. And I don't think existing calls to slice would have their performance changed much. This intermediate slice is more to simplify additional type conversions that are required to support the new `'string` and `'bytes` type specifiers.

Reply:
The performance improvements are primarily in lisp.go and env.go